src/goto-programs/interpreter_evaluate.cpp

```diff
@@ -408,10 +408,10 @@ void interpretert::evaluate(
     }
     else
     {
-      mp_integer i;
-      if(!to_integer(expr, i))
+      const auto i = numeric_cast<mp_integer>(expr);
+      if(i.has_value())
```

Comment:
could be written `if(const auto i = numeric_cast<mp_integer>(expr))` (makes clear that `i` is only used in the `if`)